jerry-core/ecma/base/ecma-helpers-conversion.c

```diff
@@ -364,7 +376,9 @@ ecma_utf8_string_to_number_by_radix (const lit_utf8_byte_t *str_p, /**< utf-8 st
  */
 ecma_number_t
 ecma_utf8_string_to_number (const lit_utf8_byte_t *str_p, /**< utf-8 string */
-                            lit_utf8_size_t str_size) /**< string size */
+                            lit_utf8_size_t str_size, /**< string size */
+                            bool allow_underscore) /**< true - if '_' characters are allowed in the string
```

Comment:
This should be an uint32_t parameter which contains the defined flag.
Then you can:
`return ecma_utf8_string_to_number_by_radix (str_p + 2, end_p, 16 | underscore_flag);`
If you really want to keep the bool, you can create a local variable which contains the flag depending on the bool.